### PR TITLE
ENH: add line numbers to selector error

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

**Comment by CJ-Wright** [comment]
@conda-forge/conda-smithy ready for review! (failing test from unrelated part of code)

**Comment by isuruf** [comment]
Can you add a test like https://github.com/conda-forge/conda-smithy/blob/master/conda_smithy/tests/test_lint_recipe.py#L135 ?

**Comment by CJ-Wright** [comment]
Done

**Comment by CJ-Wright** [comment]
@conda-forge/conda-smithy ready for review! (failing test from unrelated part of code)

**Comment by isuruf** [comment]
Thanks

**Comment by CJ-Wright** [comment]
Thank you!

**Comment by jakirkham** [comment]
Overlaps with issue ( https://github.com/conda-forge/conda-smithy/issues/105 ) and issue ( https://github.com/conda-forge/conda-smithy/issues/115 ). Would be good if we can turn lints into some proper warning as suggested in issue ( https://github.com/conda-forge/conda-smithy/issues/423 ) and start including line numbers as metadata on this warning type.

cc @goanpeca

**Comment by goanpeca** [comment]
Hmmm not much time right now, I could take a look in a couple after the 5th of October

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -104,15 +104,18 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
...
-            for selector_line in selector_lines(fh):
+            for selector_line, line_number in selector_lines(fh):
                 if not good_selectors_pat.match(selector_line):
                     bad_selectors.append(selector_line)
+                    bad_lines.append(line_number)
         if bad_selectors:
             lints.append('Selectors are suggested to take a '
-                         '``<two spaces>#<one space>[<expression>]`` form.')
+                         '``<two spaces>#<one space>[<expression>]`` form.'
+                         'see lines {}'.format(bad_lines))
```
Comment (jakirkham) [inline conda_smithy/lint_recipe.py]: Minor thing I noticed (of course after seeing in production 😄) We should add a space `form.`

xref: https://github.com/conda-forge/mosh-feedstock/pull/5#issuecomment-345473080

Comment (CJ-Wright) [reply conda_smithy/lint_recipe.py]: (A capital S would be good too!)

Comment (CJ-Wright) [reply conda_smithy/lint_recipe.py]: Also it seems to be suffering an off by one error?

**Comment by jakirkham** [comment]
Would you like to submit a follow-up PR, @CJ-Wright?

**Comment by CJ-Wright** [comment]
Sure

**Comment by CJ-Wright** [comment]
See #616

**Pull request merged**